libs/tdigest/src/main/java/org/elasticsearch/tdigest/MergingDigest.java

```diff
@@ -274,9 +271,6 @@ private void merge(
         incomingWeight.set(incomingCount, weight, 0, lastUsedCell);
         incomingCount += lastUsedCell;
 
-        if (incomingOrder == null) {
```

Comment:
This case wasn't happening, and it wasn't marked as nullable in any way, so I removed it.

I could instead add logic to close this extra array if it was created here. Opinions?